external/llvm-project/mlir/test/Target/LLVMIR/rocdl.mlir

```diff
@@ -27,7 +27,7 @@ llvm.func @rocdl_special_regs() -> i32 {
   // CHECK: call i64 @__ockl_get_num_groups(i32 2)
   %12 = rocdl.grid.dim.z : i64
 
-  // CHECK: call range(i32 0, 64) i32 @llvm.amdgcn.workitem.id.x()
+  // CHECK: call noundef range(i32 0, 64) i32 @llvm.amdgcn.workitem.id.x()
```

Comment:
Ah - this is the result of an ugly hack on our side that can be lemoved (after double-checking that `IntrinsicsAMDGPU.td` in the LLVM copy has `noundef`)

Reply:
Not yet, it seems.

Reply:
That is, take a look at the file, it'll define the intrinsics to have `noundef` as one of their inherent properties (so it wouldn't show up in the call like this)

Reply:
Many intrinsics in that file do have NoUndef<RetIndex>, but the only "workitem" definition I see is this:

  `defm int_amdgcn_workitem_id : AMDGPUReadPreloadRegisterIntrinsic_xyz;`

which has no NoUndef.  Since it's defm instead of def, I don't know if there's even a place to hang it.

Reply:
Check if we have
```
class AMDGPUReadPreloadRegisterIntrinsic
  : DefaultAttrsIntrinsic<[llvm_i32_ty], [], [NoUndef<RetIndex>, IntrNoMem, IntrSpeculatable]>;
```
above that and the comment block. Note the `NoUndef<RetIndex>`.

If that's present, we can remove our `noundef` hack

Reply:
In the file is

```
   class AMDGPUReadPreloadRegisterIntrinsic
     : DefaultAttrsIntrinsic<[llvm_i32_ty], [], [NoUndef<RetIndex>, IntrNoMem, IntrSpeculatable]>;
```

which suggests we can remove it, but on the other hand we're producing that "noundef" in the asm -- oh, is that the hack, to force it in the printer?  Either way, where is the hack that I could remove?

Reply:
Yeah. Go look in the translations to LLVM IR on `ROCDLOps.td` and the surrounding C++ source code - it'll be a patch we're carrying

Reply:
... Yeah `external/llvm-project/mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td` line 111 is where I put the hack in